shellfoundry_traffic/script_utils.py

```diff
@@ -18,15 +18,26 @@
 
 class ScriptCommandExecutor:
 
-    def __init__(self, script_definition_yaml: str) -> None:
-        shell_definition_yaml = Path(os.getcwd()).joinpath(f'{script_definition_yaml}.yaml')
-        with open(shell_definition_yaml, 'r') as file:
+    def __init__(self, script_definition: str) -> None:
+        script_definition_yaml = (script_definition if script_definition.endswith('.yaml') else
+                                  f'{script_definition}.yaml')
+        script_definition_yaml_full_path = Path(os.getcwd()).joinpath(script_definition_yaml)
+        with open(script_definition_yaml_full_path, 'r') as file:
             self.script_definition = yaml.safe_load(file)
         self.dist = Path(os.getcwd()).joinpath('dist')
         self.script_zip = self.dist.joinpath(f'{self.script_definition["metadata"]["script_name"]}.zip')
 
     def should_zip(self, file: str) -> bool:
-        return file not in self.script_definition['files']['exclude']
+        """ Returns whether the file should be added to the shell zip file or not. """
+        if 'files' in self.script_definition.keys():
+            if self.script_definition['files'] != 'exclude':
```

Comment:
There could be other entries under 'files', not just 'exclude' so I suggest you use the same method you used in the outer condition to verify if there is an 'exclude' entry inside 'files'.